File: configuration-manager-python-client/configuration_manager_client/models/files.py
"""File management models for Configuration Manager client."""

from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional, Any, Dict, List, BinaryIO, Union
from io import BytesIO

from .types import FileMetadata, FileObject


@dataclass
class ReplaceFileRequest:
    """File replacement request."""
    config_id: str
    property_path: str
    file: Union[BinaryIO, BytesIO, bytes]
    file_name: str
    content_type: Optional[str] = None


@dataclass
class UpdatedConfigInfo:
    """Updated configuration info."""
    id: str
    updated_at: datetime


@dataclass
class ReplaceFileResponse:
    """File replacement response."""
    success: bool
    message: str
    file_object: Optional[FileObject] = None
    config: Optional[UpdatedConfigInfo] = None
    error: Optional[str] = None


@dataclass
class FileInfoResponse:
    """File info response."""
    metadata: FileMetadata
    download_url: str


@dataclass
class FolderImportFile:
    """File for folder import."""
    file: Union[BinaryIO, BytesIO, bytes]
    file_name: str
    content_type: Optional[str] = None
    relative_path: Optional[str] = None


@dataclass
class FolderImportRequest:
    """Folder import request."""
    files: List[FolderImportFile]
    folder_name: Optional[str] = None
    relative_paths: Optional[List[str]] = None


@dataclass
class FolderImportStats:
    """Folder import statistics."""
    total_files: int
    json_files: int
    binary_files: int
    errors: int
    error_details: Optional[List[str]] = None


@dataclass
class FolderImportResponse:
    """Folder import response."""
    success: bool
    message: str
    data: Dict[str, Any]
    stats: FolderImportStats


@dataclass
class FileDownloadResult:
    """File download result."""
    content: bytes
    file_name: str
    content_type: str
    content_length: int

File: configuration-manager-go-client/files.go
package configmanager

import (
	"io"
	"time"
)

// ReplaceFileRequest represents file replacement request
type ReplaceFileRequest struct {
	ConfigID     string    `json:"configId"`
	PropertyPath string    `json:"propertyPath"`
	File         io.Reader `json:"-"`
	FileName     string    `json:"fileName"`
	ContentType  string    `json:"contentType,omitempty"`
}

// ReplaceFileResponse represents file replacement response
type ReplaceFileResponse struct {
	Success    bool               `json:"success"`
	Message    string             `json:"message"`
	FileObject *FileObject        `json:"fileObject,omitempty"`
	Config     *UpdatedConfigInfo `json:"config,omitempty"`
	Error      *string            `json:"error,omitempty"`
}

// UpdatedConfigInfo represents updated configuration info
type UpdatedConfigInfo struct {
	ID        string    `json:"id"`
	UpdatedAt time.Time `json:"updated_at"`
}

// FileInfoResponse represents file info response
type FileInfoResponse struct {
	Metadata    FileMetadata `json:"metadata"`
	DownloadURL string       `json:"downloadUrl"`
}

// FolderImportRequest represents folder import request
type FolderImportRequest struct {
	Files         []FolderImportFile `json:"-"`
	FolderName    *string            `json:"folderName,omitempty"`
	RelativePaths []string           `json:"relativePaths,omitempty"`
}

// FolderImportFile represents a file for folder import
type FolderImportFile struct {
	File         io.Reader `json:"-"`
	FileName     string    `json:"fileName"`
	ContentType  string    `json:"contentType,omitempty"`
	RelativePath string    `json:"relativePath,omitempty"`
}

// FolderImportResponse represents folder import response
type FolderImportResponse struct {
	Success bool                 `json:"success"`
	Message string               `json:"message"`
	Data    interface{}          `json:"data"`
	Stats   FolderImportStats    `json:"stats"`
}

// FolderImportStats represents folder import statistics
type FolderImportStats struct {
	TotalFiles   int      `json:"totalFiles"`
	JSONFiles    int      `json:"jsonFiles"`
	BinaryFiles  int      `json:"binaryFiles"`
	Errors       int      `json:"errors"`
	ErrorDetails []string `json:"errorDetails,omitempty"`
}

// FileDownloadResult represents file download result
type FileDownloadResult struct {
	Content       io.ReadCloser `json:"-"`
	FileName      string        `json:"fileName"`
	ContentType   string        `json:"contentType"`
	ContentLength int64         `json:"contentLength"`
}

// FileService provides file management operations
type FileService struct {
	client *HTTPClient
}

// NewFileService creates a new file service
func NewFileService(client *HTTPClient) *FileService {
	return &FileService{client: client}
}
